tests/matlab_test_data_collectors/matlab_collectors/collect_at_linear_array_transducer.m

```diff
@@ -0,0 +1,199 @@
+
+% create the time array
+kgrid.makeTime(c0, cfl, t_end);
+
+recorder.recordObject('kgrid', kgrid);
+
+% --------------------
+% SOURCE
+% --------------------
+
+% set indices for each element
+if rem(element_num, 2)
+    ids = (1:element_num) - ceil(element_num/2);
+else
+    ids = (1:element_num) - (element_num + 1)/2;
+end
+
+% set time delays for each element to focus at source_focus
+time_delays = -(sqrt((ids .* element_pitch).^2 + source_focus.^2) - source_focus) ./ c0;
+time_delays = time_delays - min(time_delays);
+
+% create time varying source signals (one for each physical element)
+source_sig = source_amp .* toneBurst(1/kgrid.dt, source_f0, source_cycles, 'SignalOffset', round(time_delays / kgrid.dt));
+
+% create empty kWaveArray
+karray = kWaveArray('BLITolerance', 0.05, 'UpsamplingRate', 10);
+
+% add rectangular elements
+for ind = 1:element_num
+    
+    % set element y position
+    x_pos = 0 - (element_num * element_pitch / 2 - element_pitch / 2) + (ind - 1) * element_pitch;
+    
+    % add element (set rotation angle to match the global rotation angle)
+    karray.addRectElement([x_pos, 0, kgrid.z_vec(1)], element_width, element_length, rotation);
+    
+end
+
+% move the array
+karray.setArrayPosition(translation, rotation)
+
+recorder.recordObject('karray', karray);
+% assign binary mask
+source.p_mask = karray.getArrayBinaryMask(kgrid);
+
+% assign source signals
+source.p = karray.getDistributedSourceSignal(kgrid, source_sig);
+    
+% --------------------
+% MEDIUM
+% --------------------
+
+% assign medium properties
+medium.sound_speed = c0;
+medium.density = rho0;
+
+% --------------------
+% SENSOR
+% --------------------
+
+% set sensor mask to record central plane
+sensor.mask = zeros(Nx, Ny, Nz);
+sensor.mask(:, Ny/2, :) = 1;
+
+% record the pressure
+sensor.record = {'p_max'};
+
+recorder.recordObject('sensor', sensor);
+% % --------------------
+% % SIMULATION
+% % --------------------
+
+% % set input options
+% input_args = {...
+%     'PMLSize', 'auto', ...
+%     'PMLInside', false, ...
+%     'PlotPML', false, ...
+%     'DisplayMask', 'off'};
+
+% % run code
+% switch model
+%     case 1
+        
+%         % MATLAB CPU
+%         sensor_data = kspaceFirstOrder3D(kgrid, medium, source, sensor, ...
+%             input_args{:}, ...
+%             'DataCast', 'single', ...
+%             'PlotScale', [-1, 1] * source_amp);
+        
+%     case 2
+        
+%         % MATLAB GPU
+%         sensor_data = kspaceFirstOrder3D(kgrid, medium, source, sensor, ...
+%             input_args{:}, ...
+%             'DataCast', 'gpuArray-single', ...
+%             'PlotScale', [-1, 1] * source_amp);
+        
+%     case 3
+        
+%         % C++
+%         sensor_data = kspaceFirstOrder3DC(kgrid, medium, source, sensor, input_args{:});
+        
+%     case 4
+        
+%         % C++/CUDA GPU
+%         sensor_data = kspaceFirstOrder3DG(kgrid, medium, source, sensor, input_args{:});
+        
+% end
```

Comment:
Is this supposed to be enabled at some point? If not, maybe delete?

Reply:
They're gone.